propka/output.py

```diff
@@ -1,397 +1,491 @@
-    str += "--                            BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                         --\n"
-    str += "--                                                                                                   --\n"
-    str += "--                                VERSION 3.1,  07/01/2011, COPENHAGEN                               --\n"
-    str += "--                            BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                         --\n"
-    str += "-------------------------------------------------------------------------------------------------------\n"
-    str += "\n"
-
-
-
-    return str
-
-
-def getReferencesHeader():
+    str_ = "propka3.1 {0:>93s}\n".format(today)
+    str_ += ("---------------------------------------------------------------"
+             "----------------------------------------\n")
+    str_ += ("--                                                             "
+             "                                      --\n")
+    str_ += ("--                                   PROPKA: A PROTEIN PKA "
+             "PREDICTOR                                 --\n")
+    str_ += ("--                                                             "
+             "                                      --\n")
+    str_ += ("--                                 VERSION 1.0,  04/25/2004, "
+             "IOWA CITY                               --\n")
+    str_ += ("--                                             BY HUI LI       "
+             "                                      --\n")
+    str_ += ("--                                                             "
+             "                                      --\n")
+    str_ += ("--                            VERSION 2.0,  11/05/2007, IOWA "
+             "CITY/COPENHAGEN                         --\n")
+    str_ += ("--                                BY DELPHINE C. BAS AND DAVID "
+             "M. ROGERS                             --\n")
+    str_ += ("--                                                             "
+             "                                      --\n")
+    str_ += ("--                                VERSION 3.0,  01/06/2011, "
+             "COPENHAGEN                               --\n")
+    str_ += ("--                            BY MATS H.M. OLSSON AND CHRESTEN "
+             "R. SONDERGARD                         --\n")
+    str_ += ("--                                                             "
+             "                                      --\n")
+    str_ += ("--                                VERSION 3.1,  07/01/2011, "
+             "COPENHAGEN                               --\n")
+    str_ += ("--                            BY CHRESTEN R. SONDERGARD AND "
+             "MATS H.M. OLSSON                         --\n")
+    str_ += ("---------------------------------------------------------------"
+             "----------------------------------------\n")
+    str_ += ("\n")
```

Comment:
This string-reformatting doesn't look like an improvement. How about this:
```python
    return (
        "propka3.1 {0:>93s}\n".format(today) +
        "-------------------------------------------------------------------------------------------------------\n"
        "--                                                                                                   --\n"
        "--                                   PROPKA: A PROTEIN PKA PREDICTOR                                 --\n"
        "--                                                                                                   --\n"
        "--                                 VERSION 1.0,  04/25/2004, IOWA CITY                               --\n"
        "--                                             BY HUI LI                                             --\n"
        "--                                                                                                   --\n"
        "--                            VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                         --\n"
        "--                                BY DELPHINE C. BAS AND DAVID M. ROGERS                             --\n"
        "--                                                                                                   --\n"
        "--                                VERSION 3.0,  01/06/2011, COPENHAGEN                               --\n"
        "--                            BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                         --\n"
        "--                                                                                                   --\n"
        "--                                VERSION 3.1,  07/01/2011, COPENHAGEN                               --\n"
        "--                            BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                         --\n"
        "-------------------------------------------------------------------------------------------------------\n"
        "\n")
```
Or this:
```python
    return ("propka3.1 {0:>93s}\n".format(today) + """
-------------------------------------------------------------------------------------------------------
--                                                                                                   --
--                                   PROPKA: A PROTEIN PKA PREDICTOR                                 --
--                                                                                                   --
--                                 VERSION 1.0,  04/25/2004, IOWA CITY                               --
--                                             BY HUI LI                                             --
--                                                                                                   --
--                            VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                         --
--                                BY DELPHINE C. BAS AND DAVID M. ROGERS                             --
--                                                                                                   --
--                                VERSION 3.0,  01/06/2011, COPENHAGEN                               --
--                            BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                         --
--                                                                                                   --
--                                VERSION 3.1,  07/01/2011, COPENHAGEN                               --
--                            BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                         --
-------------------------------------------------------------------------------------------------------
""")
```
Same for some similar functions in this file.

Reply:
How would you feel if I removed some of the horizontal white space as well?  It would quiet pylint a bit. :)

Reply:
Addressed in https://github.com/Electrostatics/propka-3.1/commits/nathan/strings

Reply:
Fixed in https://github.com/Electrostatics/propka-3.1/commit/b54190ef279b9acd98f6b52cfc861be14fd94e84